core/src/loader.rs

```diff
@@ -1584,7 +1585,94 @@ impl<'gc> Loader<'gc> {
+                            if let Some(v) = tag.comments().next() {
+                                id3.set_public_property(
+                                    "comment",
+                                    Avm2Value::String(AvmString::new_utf8(
+                                        activation.gc(),
+                                        v.text.clone(),
+                                    )),
+                                    &mut activation,
+                                )
+                                .expect("failed set_public_property");
+                            }
+                            if let Some(v) = tag.genre() {
+                                id3.set_public_property(
+                                    "genre",
+                                    Avm2Value::String(AvmString::new_utf8(activation.gc(), v)),
+                                    &mut activation,
+                                )
+                                .expect("failed set_public_property");
+                            }
+                            if let Some(v) = tag.title() {
+                                id3.set_public_property(
+                                    "songName",
+                                    Avm2Value::String(AvmString::new_utf8(activation.gc(), v)),
+                                    &mut activation,
+                                )
+                                .expect("failed set_public_property");
+                            }
+                            if let Some(v) = tag.track() {
+                                id3.set_public_property(
+                                    "track",
+                                    Avm2Value::String(AvmString::new_utf8(
+                                        activation.gc(),
+                                        v.to_string(),
+                                    )),
+                                    &mut activation,
+                                )
+                                .expect("failed set_public_property");
+                            }
+                            if let Some(v) = tag.year() {
+                                id3.set_public_property(
+                                    "year",
+                                    Avm2Value::String(AvmString::new_utf8(
+                                        activation.gc(),
+                                        v.to_string(),
+                                    )),
+                                    &mut activation,
+                                )
+                                .expect("failed set_public_property");
+                            }
+                        }
+
+                        sound_object
+                            .as_sound_object()
+                            .expect("Not a sound object")
+                            .set_id3(activation.context.gc_context, Some(id3));
+                        if call_id3_event {
```

Comment:
This could be written as `if tag.is_ok()` after changing the code above to `if let Ok(ref tag) = tag`.